### DISCUSSION TITLE
ModuleNotFoundError: No module named 'flask_socketio'

### DISCUSSION THREAD

**Discussion by ningelschlingel** [miguelgrinberg/Flask-SocketIO#1807] [Q&A] [answered]
Im using this on a Raspberry Pi Zero with venv.

requirements.txt:

```txt
Flask==2.0.3
Flask-SocketIO==5.1.1 # also tried flask-socketio
```

Installment:
```bash
python3 -m pip install -r requirements.txt
```

Usage:
```
from flask_socketio import SocketIO, emit
```

The error persists after running `python3 -m pip install --upgrade flask-socketio`
Installing the package with `sudo -H python3 -m pip install flask-socketio` did work for me, but that seems like an ugly workaround rather than a real solution

Thanks in advance for any hint!

**Comment by miguelgrinberg** [comment, marked as answer]
Why aren't you using a virtualenv? This is likely a mismatch between the python executable that you are using to run the application and the one that goes with the `pip` that you are using to install.

Using a virtual environment removes all these side effects.

**Reply by ningelschlingel** [reply]
Actually I do,  as follows:

``` bash
python3 -m venv "{path to project}/venv"
. {path to project}/venv/bin/activate
python -m pip install -r requirements.txt
```
(Was just giving the minimal example)

I tried a lot and at this point I don't have any idea how this error even occurs after a successful installation with pip.
The sd-card for the pi was freshly set up with Pi Imager to be able to rule out old interfering packages as cause.

Thanks for this insanely fast answer!

**Reply by miguelgrinberg** [reply]
Okay, good on the virtualenv. How are you running your application?

**Reply by ningelschlingel** [reply]
That was the key ... or better to say is.

I have to run `sudo python3 app.py` because im using gpio which requires root permission.
Unfortunately my earlier tests did not lead me to this conclusion ... im running it through a script and never really thought about it at this point ... my bad.

many thanks for your time!

Edit: Still wondering why Flask and other dependencies don't lead to the same error, since im importing them before flask-socketio, and as I said I used a freshly build sc-card. But I don't imagine that's related to this package

**Reply by ningelschlingel** [reply]
Turns out, that even after a fresh Raspberry Pi OS Build, Flask is installed. 
I don't know if its preinstalled, or if the folder/package/etc somehow survived the process, but that's what led me on a wrong track
